src/components/Photo/Selfie.js

```diff
@@ -11,27 +11,108 @@ import style from './style.css'
 
 type State = {
   hasBecomeInactive: boolean,
+  snapshotBuffer: Array<{
+    blob: Blob,
+    base64: string
+  }>,
 }
 
 type Props = {
   onCapture: Function,
   renderFallback: Function,
   trackScreen: Function,
+  useMultipleSelfieCapture: boolean,
+  snapshotInterval: number,
 }
 
 const inactiveError = { name: 'CAMERA_INACTIVE', type: 'warning' }
 
 export default class Selfie extends Component<Props, State> {
   webcam = null
+  snapshotIntervalRef: ?IntervalID = null
 
   state: State = {
     hasBecomeInactive: false,
+    snapshotBuffer: [],
   }
 
   handleTimeout = () => this.setState({ hasBecomeInactive: true })
 
-  handleClick = () => screenshot(this.webcam,
-    (blob, base64) => this.props.onCapture({ blob, base64 }))
+  /* When taking snapshots we want to ensure that the snapshot we provide is, where possible,
+    older than the interval. As we take a snapshot every interval, that means we need to
+    store the previous snapshot so that we can use that if the capture is taken too soon
+    after the latest snapshot (e.g. 500ms after the last snapshot on a 1000ms interval).
+    
+    Therefore we use a simple two item collection as our buffer:
+    
+    ['ready snapshot', 'fresh snapshot']
+
+    The first item is the older of the two snapshots, and will be used when capturing the selfie.
+    The second item is the 'fresh' slot which will not be suitable until the next interval tick at which
+    point it is moved to the 'ready' slot. If there are no items in the buffer, then
+    the new snapshot will be placed in both positions in the buffer, so that we always provide a
+    snapshot (even if that could sometimes be a suboptimal time delta) */
+  takeSnapshot = () => {
+    screenshot(this.webcam, (blob, base64) => {
+      this.setState(({ snapshotBuffer: [, newestSnapshot] }) => ({
+        snapshotBuffer: [
+          ...([newestSnapshot] || [
```

Comment:
this part is a bit confusing, can we actually get an array with two equal values?

Reply:
I'm going to neaten this up - the idea is to always have a snapshot available, even if there isn't an ideal 'mature' snapshot which is far enough in the past to be a good comparison. However, this can be handled with less faff when selecting the snapshot during the onClick.